File: Array/217-Contains-Duplicate.java
import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        int len1 = nums.length;
        Arrays.sort(nums);
        for (int i = 0; i < len1 - 1; i++) {
            if (nums[i] == nums[i + 1]) {
                return true;
            }
        }

        return false;
    }

    public boolean containsDuplicate1(int[] nums) {
        Set<Integer> numSet = new HashSet<>();

        for (int num : nums) {
            if (numSet.contains(num)) {
                return true;
            }

            numSet.add(num);
        }
        return false;
    }
}

File: Array/217-Contains-Duplicate.ts
function containsDuplicate(nums: number[]): boolean {
    let numSet: Set<number> = new Set();
    // A solution without sorting may be faster
    for (const num of nums) {
        if (numSet.has(num)) {
            return true;
        }
        numSet.add(num);
    }
    return false;
};

function containsDuplicate1(nums: number[]): boolean {
    const len = nums.length;
    const newNums = nums.sort();
    for (let i = 0; i < len; i++) {
        if (newNums[i] == nums[i+1]) {
            return true;
        }        
    }
    return false;
};